src/dfx/src/lib/builders/mod.rs

```diff
@@ -118,16 +118,6 @@ pub trait CanisterBuilder {
                 )
             })?;
         }
-        std::fs::create_dir_all(&generate_output_dir).with_context(|| {
```

Comment:
This code was generating the declarations directory and the DID file before checking if the bindings array is empty.
There's some code lower down in this file that deletes the DID file, but we never get that far because of the early return for an empty bindings array.

So the old flow was like this:
- create directory
- create did file
- if bindings array is empty, return early
- create JS bindings if applicable
- create TS bindings if applicable
- delete DID bindings if not applicable

Now it's like this:
- if bindings array is empty, return early
- create directory
- create did file
- create JS bindings if applicable
- create TS bindings if applicable
- delete DID bindings if not applicable